Fixes SIZE_TYPE to be "unsigned int" instead of "long unsigned int".

This makes size_t to be "unsigned" ssize_t which makes happy compiler on data
type checks.

Fix is taken from current development branch of GCC for ARC and will be a
part of the next release of ARC tools, so at that point patch should be dropped.

https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/commit/249f040299402647525c3f15b79d319fa7acddd3

Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
---

--- a/gcc/config/arc/arc.h
+++ b/gcc/config/arc/arc.h
@@ -487,7 +487,7 @@ if (GET_MODE_CLASS (MODE) == MODE_INT		\
 /* Define this as 1 if `char' should by default be signed; else as 0.  */
 #define DEFAULT_SIGNED_CHAR 0
 
-#define SIZE_TYPE "long unsigned int"
+#define SIZE_TYPE "unsigned int"
 #define PTRDIFF_TYPE "long int"
 #define WCHAR_TYPE "int"
 #define WCHAR_TYPE_SIZE 32
